themes/bootstrap3/js/truncate.js

```diff
@@ -0,0 +1,188 @@
+/* global VuFind */
+
+VuFind.register('truncate', function Truncate() {
+  const defaultSettings = {
+    'btn-class': '',
+    'in-place-toggle': false,
+    'label': null,
+    'less-label': VuFind.translate('show_less'),
+    'more-label': VuFind.translate('show_more'),
+    'rows': 3,
+    'top-toggle': Infinity,
+    'wrapper-class': '', // '' will glean from element, false or null will exclude a class
+    'wrapper-tagname': null, // falsey values will glean from element
+  };
+
+  function initTruncate(_container, _element, _fill) {
+    var zeroHeightContainers = [];
+
+    $(_container).not('.truncate-done').each(function truncate() {
+      var container = $(this);
+      var settings = Object.assign({}, defaultSettings, container.data('truncate'));
+
+      var element = typeof _element !== 'undefined'
+        ? container.find(_element)
+        : (typeof settings.element !== 'undefined')
+          ? container.find(settings.element)
+          : false;
+      var fill = typeof _fill === 'undefined' ? function fill(m) { return m; } : _fill;
+      var rowCount = settings.rows;
+      var moreLabel, lessLabel;
+      moreLabel = lessLabel = settings.label;
+      if (moreLabel === null) {
+        moreLabel = settings['more-label'];
+        lessLabel = settings['less-label'];
+      }
+      var btnClass = settings['btn-class'] ? ' ' + settings['btn-class'] : '';
+      var topToggle = settings['top-toggle'];
+      var inPlaceToggle = (element && settings['in-place-toggle'])
+        ? settings['in-place-toggle']
+        : false;
+
+      var parent, numRows, shouldTruncate, truncatedHeight;
+      var wrapperClass = settings['wrapper-class'];
+      var wrapperTagName = settings['wrapper-tagname'];
+      var toggleElements = [];
+
+      if (element) {
+        // Element-based truncation
+        parent = element.parent();
+        numRows = container.find(element).length || 0;
+        shouldTruncate = rowCount < numRows;
+
+        if (wrapperClass === '') {
+          wrapperClass = element.length ? element.prop('class') : '';
+        }
+        if (!wrapperTagName) {
+          wrapperTagName = element.length && element.prop('tagName').toLowerCase();
+        }
+
+        if (shouldTruncate) {
+          element.each(function hideRows(i) {
+            if (i === rowCount) {
+              $(this).addClass('truncate-start');
+            }
+            if (i >= rowCount) {
+              $(this).hide();
+              $(this).addClass('truncate-toggle');
```

Comment:
Maybe we can leave this out, unless we want to provide a class expressly for styling togglable elements? If we do want to provide a class, maybe it could be renamed to `truncate-togglable`?

Reply:
I don't have strong feelings about this; I left it in just in case you were already relying on it for styling in your existing implementation, but if you prefer to change or remove it, please feel free.